lms/static/js/learner_dashboard/models/program_subscription_model.js

```diff
@@ -0,0 +1,87 @@
+import Backbone from 'backbone';
+import moment from 'moment';
+
+import DateUtils from 'edx-ui-toolkit/js/utils/date-utils';
+
+/**
+ * Model for Program Subscription Data.
+ */
+class ProgramSubscriptionModel extends Backbone.Model {
+    constructor({ context }, ...args) {
+        const {
+            subscriptionData: [data = {}],
+            programData: { subscription_prices },
+            urls,
+            userPreferences,
+        } = context;
+
+        const subscriptionState = data.subscription_state?.toLowerCase() ?? '';
+        const subscriptionPrice = '$' + parseFloat(
+            subscription_prices?.find(({ currency }) => currency === 'USD')?.price
+        );
+        const subscriptionUrl =
+            subscriptionState === 'active'
+                ? urls.manage_subscription_url
+                : urls.buy_subscription_url;
+
+        const hasActiveTrial =
+            subscriptionState === 'active' && data.trial_end
+                ? moment(
+                    DateUtils.localizeTime(
+                        DateUtils.stringToMoment(data.trial_end),
+                        'UTC'
+                    )
+                ).isAfter(moment.utc())
+                : false;
```

Comment:
nit: another place where I found this to be a bit hard to read\grok because we're doing so much in a ternary expression.

Reply:
sure, will make it more readable :)